example/src/main/java/co/infinum/example/MainActivity.kt

```diff
@@ -6,106 +6,163 @@ import android.content.pm.PackageManager
-        AppCompatDelegate.setCompatVectorFromResourcesEnabled(true)
-        initGoldenEye()
+        setContentView(R.layout.main_activity2)
+        if (allPermissionsGranted()) {
+            startCamera()
+        } else {
+            ActivityCompat.requestPermissions(
+                this, REQUIRED_PERMISSIONS, REQUEST_CODE_PERMISSIONS
+            )
+        }
+
+        outputDirectory = getOutputDirectory()
+        cameraExecutor = Executors.newSingleThreadExecutor()
         videoFile = File.createTempFile("vid", "")
+
         initListeners()
+        setUpPinchToZoom()
     }
 
     private fun initListeners() {
-        settingsView.setOnClickListener {
-            prepareItems()
-            settingsRecyclerView.apply {
-                visibility = View.VISIBLE
-                layoutManager = LinearLayoutManager(this@MainActivity)
-                adapter = settingsAdapter
-            }
-        }
+//        settingsView.setOnClickListener {
+//            prepareItems()
+//            settingsRecyclerView.apply {
+//                visibility = View.VISIBLE
+//                layoutManager = LinearLayoutManager(this@MainActivity2)
+//                adapter = settingsAdapter
+//            }
+//        }
 
         takePictureView.setOnClickListener { _ ->
-            goldenEye.takePicture(
+            takePicture(
                 onPictureTaken = { bitmap ->
                     if (bitmap.width <= 4096 && bitmap.height <= 4096) {
                         displayPicture(bitmap)
                     } else {
                         reducePictureSize(bitmap)
                     }
-                },
-                onError = { it.printStackTrace() }
+                }
             )
         }
 
         recordVideoView.setOnClickListener { _ ->
             if (isRecording) {
                 isRecording = false
                 recordVideoView.setImageResource(R.drawable.ic_record_video)
-                goldenEye.stopRecording()
+                videoCapture?.stopRecording()
             } else {
                 startRecording()
             }
         }
 
         switchCameraView.setOnClickListener { _ ->
-            val currentIndex = goldenEye.availableCameras.indexOfFirst { goldenEye.config?.id == it.id }
-            val nextIndex = (currentIndex + 1) % goldenEye.availableCameras.size
-            openCamera(goldenEye.availableCameras[nextIndex])
+            lensFacing =
+                if (lensFacing == CameraSelector.LENS_FACING_BACK) CameraSelector.LENS_FACING_FRONT else CameraSelector.LENS_FACING_BACK
+            CameraX.getCameraWithLensFacing(lensFacing)
+            recreateCamera()
         }
     }
 
+    private fun recreateCamera() {
+        CameraX.unbindAll()
+        startCamera()
+    }
+
+    private fun takePicture(onPictureTaken: (Bitmap) -> Unit) {
+        // Get a stable reference of the modifiable image capture use case
+        val imageCapture = imageCapture ?: return
```

Comment:
Effectively the same as just doing `imageCapture?.takePicture(...)` below, which I prefer since that's the only use-case.